basic_bootloader/src/bootloader/runner.rs

```diff
@@ -754,6 +813,7 @@ where
             deployment_success
         ));
 
+        resources_returned.reclaim(resources_for_deployer);
```

Comment:
I agree this isn't nice, but it goes beyond the name of the function (we can change it to + if you want). Ideally, this adding back of returned resources should only happen at 1 point instead of being scattered around. 
Maybe we can pass resources_to_give_back (from prep) to the EE from launch, and then the result will have a unified resource to rejoin with caller's?